services/scabbard/libscabbard/src/service/factory.rs

```diff
@@ -34,7 +34,92 @@ const DEFAULT_STATE_DB_SIZE: usize = 1 << 30; // 1024 ** 3
 const DEFAULT_RECEIPT_DB_DIR: &str = "/var/lib/splinter";
 const DEFAULT_RECEIPT_DB_SIZE: usize = 1 << 30; // 1024 ** 3
 
-#[cfg(feature = "factory-builder")]
+#[cfg(not(feature = "diesel-receipt-store"))]
+type ScabbardReceiptStore = Arc<RwLock<TransactionReceiptStore>>;
+
+#[cfg(feature = "diesel-receipt-store")]
+type ScabbardReceiptStore = Arc<RwLock<dyn ReceiptStore>>;
+
+#[cfg(feature = "diesel-receipt-store")]
+pub enum ConnectionUri {
+    #[cfg(feature = "postgres")]
+    Postgres(Box<str>),
+    #[cfg(feature = "sqlite")]
+    Sqlite(Box<str>),
+}
+
+/// Configuration for underlying storage that will be enabled for each service produced by the
+/// resulting ScabbardFactory.
+pub enum ScabbardStorageConfiguration {
+    Lmdb {
+        db_dir: Option<String>,
+        db_size: Option<usize>,
+    },
+    #[cfg(feature = "diesel-receipt-store")]
+    DatabaseConnectionUri { connection_uri: ConnectionUri },
+}
+
+impl ScabbardStorageConfiguration {
+    fn lmdb() -> Self {
+        ScabbardStorageConfiguration::Lmdb {
+            db_dir: None,
+            db_size: None,
+        }
+    }
+
+    #[cfg(all(
+        feature = "diesel-receipt-store",
+        any(feature = "postgres", feature = "sqlite")
+    ))]
+    fn connection_uri(connection_uri: &str) -> Self {
+        let connection_uri =
+            if cfg!(feature = "postgres") && connection_uri.starts_with("postgres://") {
+                #[cfg(feature = "postgres")]
+                {
+                    ConnectionUri::Postgres(connection_uri.into())
+                }
+                // This cannot be reached do to the if test failing without this feature.
```

Comment:
The comment is confusing, please reword it

Reply:
Repaced this with an Unknown type when sqlite is not enabled, which simplifies conversion with a  match statement